Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -789,7 +789,9 @@ respondQuote (driverId, _) req = do
       case req.response of
         Pulled -> throwError UnexpectedResponseValue
         Accept -> do
-          when (searchReq.autoAssignEnabled == Just True) $ CS.incrementSearchRequestLockCounter searchTryId
+          when (searchReq.autoAssignEnabled == Just True) $
```

Comment:
this way how are we handling if any driver have already responded then cancel search should not work in autoassign flow.

Reply:
> this way how are we handling if any driver have already responded then cancel search should not work in autoassign flow.

just checking the lock

Reply:
@DonDokidon Previously, we were checking if driver has already given a quote then search cancel should fail or if search has cancelled and then driver gave the quote then driver quote should fail. 

But with your current implementation, if search has cancelled then driver can't give quote, or search can't be cancelled again.
But it didn't cover the case, when driver has given the quote then search cancelled shouldn't work.

Reply:
> But with your current implementation, if search has cancelled then driver can't give quote, or search can't be cancelled again. But it didn't cover the case, when driver has given the quote then search cancelled shouldn't work.

Fixed